core/types/block.go

```diff
@@ -82,6 +82,7 @@ type Header struct {
 	Extra       []byte         `json:"extraData"        gencodec:"required"`
 	MixDigest   common.Hash    `json:"mixHash"`
 	Nonce       BlockNonce     `json:"nonce"`
+	IsEuclid    bool           `json:"isEuclid" rlp:"optional"`
```

Comment:
This field will be serialized if `true`. Does it also mean that it will be part of the hash? If so then we can't have this field

Reply:
This is not needed anymore. I have a new rlp:"optional" field but it will not be part of the hash.